sql/src/main/java/io/crate/executor/transport/ExecutionNodesTask.java

```diff
@@ -299,6 +333,10 @@ private RamAccountingContext trackOperation(ExecutionNode executionNode, String
 
     @Override
     public List<ListenableFuture<TaskResult>> result() {
+        List<ListenableFuture<TaskResult>> results = new ArrayList<>(this.results.size());
```

Comment:
unfortunately it will not compile otherwise, don't understand this exactly cause SettableFuture implements ListenableFuture... wanted to discuss this anyway, thx for pointing out